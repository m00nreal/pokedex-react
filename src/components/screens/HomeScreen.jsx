import React, {useEffect, useState} from 'react';
import PokedexList from "../pokedex/PokedexList";
import {useFetch} from "../../hooks/useFetch";
import ReactPaginate from 'react-paginate';

const HomeScreen = () => {

    // Initial state
    const [state, setState] = useState({
        currentPage: 0,
        offset: 0,
        perPage: 24
    });

    // Get pokemons
    const { data, loading } = useFetch(`https://pokeapi.co/api/v2/pokemon?offset=${state.offset}&limit=${state.perPage}`);

    const orderedData = !loading && data.sort((a, b) => {
        return a.id - b.id;
    })

    // Pagination
    const handlePageClick = ( newPage ) => {
        const newOffset = state.perPage * newPage.selected;
        setState({
            ...state,
            offset: newOffset,
            currentPage: newPage.selected
        })
    };

    return (
        <div className="container">
            <h1 className="title text-center text-primary mt-4">All PoKeMoN</h1>
            <hr/>
            <PokedexList data={orderedData} loading={loading}/>
            <div className="d-none d-md-flex justify-content-center paginate">
                <ReactPaginate
                    previousLabel="<"
                    nextLabel=">"
                    breakLabel="..."
                    pageCount={41}
                    marginPagesDisplayed={7}
                    pageRangeDisplayed={7}
                    onPageChange={handlePageClick}
                    containerClassName="pagination"
                    pageClassName={'pagination-list-item'}
                    pageLinkClassName={'link'}
                    activeClassName={'active-link'}
                    activeLinkClassName={'active-link'}
                />
            </div>
            {/*hidden on larger screens. on smaller devices it has a lower pagination range*/}
            <div className="d-md-none d-block paginate">
                <ReactPaginate
                    previousLabel="<"
                    nextLabel=">"
                    breakLabel="..."
                    pageCount={41}
                    marginPagesDisplayed={3}
                    pageRangeDisplayed={2}
                    onPageChange={handlePageClick}
                    ontainerClassName="pagination"
                    pageClassName={'pagination-list-item'}
                    pageLinkClassName={'link'}
                    activeClassName={'active-link'}
                    activeLinkClassName={'active-link'}
                />
            </div>
        </div>
    );
};

export default HomeScreen;
