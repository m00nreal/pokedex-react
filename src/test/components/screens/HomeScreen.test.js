import HomeScreen from "../../../components/screens/HomeScreen";
import React from "react";
import {shallow} from "enzyme";

describe('Test in <HomeScreen/>', function () {

    it('should match snapshot', function () {

        const wrapper = shallow(<HomeScreen/>)
        expect(wrapper).toMatchSnapshot();
    });

});