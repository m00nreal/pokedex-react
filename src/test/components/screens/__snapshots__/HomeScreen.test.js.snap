// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <HomeScreen/> should match snapshot 1`] = `
<div
  className="container"
>
  <h1
    className="title text-center text-primary mt-4"
  >
    All PoKeMoN
  </h1>
  <hr />
  <PokedexList
    data={false}
    loading={true}
  />
  <div
    className="d-none d-md-flex justify-content-center paginate"
  >
    <PaginationBoxView
      activeClassName="active-link"
      activeLinkClassName="active-link"
      breakLabel="..."
      containerClassName="pagination"
      disableInitialCallback={false}
      disabledClassName="disabled"
      marginPagesDisplayed={7}
      nextClassName="next"
      nextLabel=">"
      onPageChange={[Function]}
      pageClassName="pagination-list-item"
      pageCount={41}
      pageLinkClassName="link"
      pageRangeDisplayed={7}
      previousClassName="previous"
      previousLabel="<"
    />
  </div>
  <div
    className="d-md-none d-block paginate"
  >
    <PaginationBoxView
      activeClassName="active-link"
      activeLinkClassName="active-link"
      breakLabel="..."
      disableInitialCallback={false}
      disabledClassName="disabled"
      marginPagesDisplayed={3}
      nextClassName="next"
      nextLabel=">"
      onPageChange={[Function]}
      ontainerClassName="pagination"
      pageClassName="pagination-list-item"
      pageCount={41}
      pageLinkClassName="link"
      pageRangeDisplayed={2}
      previousClassName="previous"
      previousLabel="<"
    />
  </div>
</div>
`;
